ivadomed/loader/mri3d_subvolume_segmentation_dataset.py

```diff
@@ -96,14 +101,21 @@ def _prepare_indices(self):
             for x in range(0, (shape[0] - self.length[0]) + 1, self.stride[0]):
                 for y in range(0, (shape[1] - self.length[1]) + 1, self.stride[1]):
                     for z in range(0, (shape[2] - self.length[2]) + 1, self.stride[2]):
-                        self.indexes.append({
-                            'x_min': x,
-                            'x_max': x + self.length[0],
-                            'y_min': y,
-                            'y_max': y + self.length[1],
-                            'z_min': z,
-                            'z_max': z + self.length[2],
-                            'handler_index': i})
+                        x_min, x_max = x, x + self.length[0]
+                        y_min, y_max = y, y + self.length[1]
+                        z_min, z_max = z, z + self.length[2]
+
+                        subvolume = {
+                            'input': list(np.asarray(input_img)[:, x_min: x_max, y_min: y_max, z_min: z_max]),
+                            'gt': list(np.asarray(gt)[:, x_min: x_max, y_min: y_max, z_min: z_max] if gt else []),
```

Comment:
I would remove the space after the colon to avoid confusion with a key-value pair.
```suggestion
                            'input': list(np.asarray(input_img)[:, x_min:x_max, y_min:y_max, z_min:z_max]),
                            'gt': list(np.asarray(gt)[:, x_min:x_max, y_min:y_max, z_min:z_max] if gt else []),
```

Reply:
Is the conversion to `list` required here? To be tested, but it seems like keeping it as an array would have the same result. Then, I would suggest the following version. 
```suggestion
                            'input': np.asarray(input_img)[:, x_min:x_max, y_min:y_max, z_min:z_max],
                            'gt': np.asarray(gt)[:, x_min:x_max, y_min:y_max, z_min:z_max] if gt else [],
```

Reply:
The first point is addressed with [576858e](https://github.com/ivadomed/ivadomed/pull/1164/commits/576858e7de5a3bdf9f0bd8b9a035dd3c7307cd0e).

For the second point, this is also how it is done in `loader/mri2d_segmentation_dataset.py`:

https://github.com/ivadomed/ivadomed/blob/e5f14c02a6c73d9360eee130ff39f0a364e0a697/ivadomed/loader/mri2d_segmentation_dataset.py#L203-L204

Not sure if there is a reason to keep this as a `list` 🤷‍♂️

Reply:
Yeah, I know it is present at different places, I'll investigate.